src/tls/connection.c

```diff
@@ -790,10 +791,18 @@ connection_crypto_init (const char *certfile,
   if (ret != GNUTLS_E_SUCCESS)
     errx (EXIT_FAILURE, "gnutls_certificate_allocate_credentials failed: %s", gnutls_strerror (ret));
 
-  if (keyfile)
-    ret = gnutls_certificate_set_x509_key_file (parameters.x509_cred, certfile, keyfile, GNUTLS_X509_FMT_PEM);
-  else
-    ret = set_x509_key_from_combined_file (parameters.x509_cred, certfile);
+  /* check if we have a separate key file */
+  keyfile = cockpit_certificate_key_path (certfile);
+  assert (keyfile);
+  ret = gnutls_certificate_set_x509_key_file (parameters.x509_cred, certfile, keyfile, GNUTLS_X509_FMT_PEM);
+
+  /* if not, fall back to combined file */
+  if (ret == GNUTLS_E_FILE_ERROR)
```

Comment:
hm.  This could also be a problem with the cert file, but I guess we'll see the same error again when we try to load the combined file...

Reply:
Right, then we'll get the error message from the "combined" fallback.